### PR TITLE
Add seperate runner for Discovery global search

### PR DESCRIPTION
**Pull request by AshTNA**
Hi @gtvj 

Would you be able to check this PR and merge if all is OK?

I noticed on Discovery, "Search for records" should be the default search type on Global Search. This is how the legacy global search behaves. In order to get this functionality  to  work, I had to create a separate HTML id for React to hook onto, and that runner has a prop which tells react to show "Search for records" by default.

This PR also fixes React crashing on certain Discovery pages such as the homepage where Global Search doesn't exist. Unlike TNA/TNA-Base, etc., Discovery does not show global search across the application, but the runner was trying to render Global Search regardless of the HTML element's existence.

There's some slight CSS issues with the mobile search spacing on Discovery, but this is a Discovery CSS issue, not a component CSS issue so I will fix it in Discovery.

Thanks :+1:

### CODE REVIEW

File: src/runners/global-search-runner.js [hunk 1]
```diff
@@ -4,12 +4,28 @@ import GlobalSearch from '../../components/global_search/global-search.react';
...
+function renderIfExists(id, desktop, defaultFormDataIndex) {
+
+	if(document.getElementById(id)) {
+		ReactDOM.render(
+			<GlobalSearch desktop={desktop} defaultFormDataIndex={defaultFormDataIndex}
+			/>,
+			document.getElementById(id)
+		)
+	}
+
```
Comment (gtvj) [inline src/runners/global-search-runner.js:17]: I wonder if we should remove these two additional line breaks - just a minor formatting thing.

File: src/runners/global-search-runner.js [hunk 2]
```diff
@@ -4,12 +4,28 @@ import GlobalSearch from '../../components/global_search/global-search.react';
...
+// Non-discovery runner
+
+renderIfExists("global-search-desktop-component", true, 0);
+renderIfExists("global-search-mobile-component", false, 0);
+
+
+// Discovery runner
+
+renderIfExists("global-search-desktop-component-discovery", true, 1);
+renderIfExists("global-search-mobile-component-discovery", false, 1);
```
Comment (gtvj) [inline src/runners/global-search-runner.js:31]: I wonder if we should add a newline here (just for consistency and to get rid of the GitHub warning icon)

**Review by gtvj — CHANGES_REQUESTED** [review]
Hi @AshTNA - I've just requested some very minor formating changes. Once these have been made I'll merge it

**Comment by AshTNA** [comment]
Thanks @gtvj - I have made those changes :+1:

**Pull request merged**